index.html

```diff
@@ -2719,6 +2761,152 @@ <h2>
           </dd>
         </dl>
       </section>
+      <section data-dfn-for="AddressErrorFields" data-link-for=
+      "AddressErrorFields">
+        <h2>
+          <dfn>AddressErrorFields</dfn> dictionary
+        </h2>
+        <pre class="idl">
+          dictionary AddressErrorFields {
+            DOMString addressLineError;
```

Comment:
Decided to postfix all these with "Error" in the name, as I felt it made it a little bit more clear when in userland code.

Reply:
The extra "Error" suffix doesn't feels redundant me, as it should be implied since these are members of the shippingAddressErrors property.

Reply:
depends... the problem I ran into was in destructuring and using object short hands. Also, consider the `shippingAddressErrors` can be the result of another function. Thus:

```JS
function validateAddress(address){
   // whoops, now "postalCode" has been squatted. 
   const  { postalCode } = address;
   const postalCodeError =  !isValidPostCode(postCode) ? "Invalid zip code" : undefined;
   return { postalCodeError }
}

function collectErrors(response) {
   // same problem here!
   const { shippingAddress } = response;
   const  shippingAddressErrors = validateAddress(shippingAddress);
   return { shippingAddressErrors }
}
```

And so on...

Reply:
Fair enough :)

Reply:
I tend to agree this is redundant. I think you have chosen a code style with a combination of object literal shorthand, destructuring, and redundant variable declarations that makes it extra confusing. But I don't think that's a good reason to verbose-ify the API, personally. I would instead suggest writing your code in a more conventional way, such as

```js
function validateAddress(address){
   return {
    postalCode: !isValidPostalCode(address.postalCode) ? "Invalid zip code" : undefined
  };
}

function collectErrors(response) {
   return {
    shippingAddress: validateAddress(response.shippingAddress)
  };
}
```

(basically, don't declare variables you're only going to use once.